packages/compass-components/src/components/guide-cue/guide-cue-service.ts

```diff
@@ -88,14 +88,21 @@ export class GuideCueService extends EventTarget {
     if (!this._activeCue) {
       return;
     }
-    return this.dispatchEvent(
-      new CustomEvent('show-cue', {
-        detail: {
-          cueId: this._activeCue.cueId,
-          groupId: this._activeCue.groupId,
-        },
-      })
-    );
+    try {
+      return this.dispatchEvent(
+        new CustomEvent('show-cue', {
+          detail: {
+            cueId: this._activeCue.cueId,
+            groupId: this._activeCue.groupId,
+          },
+        })
+      );
+    } catch (ex) {
```

Comment:
What error?

But also looking at lime 89 it looks like this function doesn't return anything if it early returns and this case should probably do the same?

Reply:
For some reason, in some tests, it throws an error trying to map the error and dispatch it. It looks like a race condition, that doesn't really affect the test behaviour, but when it bubbles up, stops the test.

And you are right about line 89, I can just remove the return and leave it return nothing. Fixing it.

Reply:
I'm just wondering if that's not best fixed in a test rather than the code under test.

Reply:
It seems to happen on all tests that depend on this service. What I feel it's that there is some race condition between the service being ready and dispatching the event, failing sometimes.

I feel that fixing the underlying issue would be expensive and it doesn't seem to affect production code.

Reply:
OK That's fair. Let's file a ticket for that and then you can leave a `// TODO(ticket-number): blah` style comment here and merge.

Reply:
Makes sense, will do. Thanks :heart: